lib/system: More Types of CPUs Added

vincrusher opened the pull request:
I have added more types of CPUs to try to support.

These could be useful for porting Nix packages to more platforms.

vincrusher committed:
Update parse.nix

Mic92 commented on lib/systems/parse.nix:
```diff
@@ -80,6 +80,7 @@ rec {
     armv8r   = { bits = 32; significantByte = littleEndian; family = "arm"; version = "8"; };
     armv8m   = { bits = 32; significantByte = littleEndian; family = "arm"; version = "8"; };
     aarch64  = { bits = 64; significantByte = littleEndian; family = "arm"; version = "8"; };
+    aarch64_be = { bits = 64; significantByte = bigEndian; family = "arm" version = "8"; };
```
```suggestion
    aarch64_be = { bits = 64; significantByte = bigEndian; family = "arm"; version = "8"; };
```

Mic92 committed:
CPU Fix

Co-Authored-By: vincrusher <vincentweisner@icloud.com>

xeji commented on lib/systems/parse.nix:
```diff
@@ -101,6 +103,8 @@ rec {
     wasm32   = { bits = 32; significantByte = littleEndian; family = "wasm"; };
     wasm64   = { bits = 64; significantByte = littleEndian; family = "wasm"; };
+    
+    alpha    = { bits = 64; significantBye = littleEndian; family = "alpha"; };
```
typo: `significantByte` to fix eval

vincrusher committed:
Update parse.nix

vincrusher committed:
lib/system: Added Embedded Platforms for the CPUs

I added some embedded platforms for the CPUs I added to the parse.nix file.
These could be used as new platforms for the added CPUs.

vincrusher committed:
lib/systems: Added missing semicolons

Mic92 commented on lib/systems/examples.nix:
```diff
@@ -112,11 +112,26 @@ rec {
     libc = "newlib";
   };
+  
+  aarch64be-embedded = {
```
@Ericson2314 do we have any requirements for systems added in this file?

Ericson2314 replied:
Not really! The only thing would be trying to keep the `config`s unique.

matthewbauer commented:
Yes- the parse.nix changes are good but the changes to examples.nix might require some more discussion. Adding "alpha-embedded" definitely seems good because none of the other ones are alpha but I'm not so sure about the power/aarch64 "endian" cases. We don't need to cover every possible combination here - just the most common ones. I think both of those processors can switch between the two.

vincrusher commented:
They do support big-endian and little-endian. I thought adding support for both as some people might like to use little over big or vice versa.

matthewbauer commented:
Looks fine for now!
